src/net.cpp

```diff
@@ -1022,7 +1055,7 @@ bool CConnman::AttemptToEvictConnection()
                                                HasAllDesirableServiceFlags(node->nServices),
                                                peer_relay_txes, peer_filter_not_null, node->nKeyedNetGroup,
                                                node->m_prefer_evict, node->addr.IsLocal(),
-                                               node->m_inbound_onion};
+                                               node->m_inbound_onion, node->addr.IsI2P()};
```

Comment:
> What about passing `enum Network` instead?

Good idea, looks like we can do that easily and it will include future network additions:

```diff
-                                               node->m_inbound_onion, node->addr.IsI2P()};
+                                               node->ConnectedThroughNetwork()};
```

Reply:
Requires a lot of changes but probably still worth it. Will look at replacing `m_is_local` as well, maybe giving it a `Network` value of `NET_UNROUTABLE` for the eviction logic.

Reply:
Passing `enum Network`, instead of multiple booleans, indeed allowed rewriting this to be better abstracted and generalizable. WIP, for now the initial rewrite is in an additional commit.